packages/grid/x-data-grid/src/hooks/features/columns/useGridColumns.tsx

```diff
@@ -112,6 +112,17 @@ export function useGridColumns(
     [apiRef],
   );
 
+  const getColumns = React.useCallback<GridColumnApi['getColumns']>(
+    (fields) => {
+      if (!Array.isArray(fields) || fields.length === 0) {
+        return [];
+      }
```

Comment:
This is too defensive, we use TS to typecheck, if the user doesn't pass an array it's on them. And the `fields.length === 0` case is handled by the code below without substantial cost, so this branch can be pruned.